tests/test_discovered_resources.py

```diff
@@ -209,26 +278,85 @@ async def test_discovered_resource_get_paging(server, client, agent, environment
     response = await http_client.fetch(request, raise_error=False)
     assert response.code == 200
     response = json.loads(response.body.decode("utf-8"))
-    assert response["data"] == discovered_resources[0:2]
+    check_expected_result(discovered_resources[0:2], response["data"])
     assert response["links"].get("prev") is None
     assert response["links"].get("next") is not None
     assert response["metadata"] == {"total": 6, "before": 0, "after": 4, "page_size": 2}
 
 
 async def test_discovery_resource_bad_res_id(server, client, agent, environment):
     """
-    Test that exceptions are raised when creating discovered resources with invalid IDs.
+    Test that exceptions are raised when creating discovered resources with invalid resource IDs.
+    Tests both creation endpoints (`discovered_resource_create` and `discovered_resource_create_batch`)
     """
     result = await agent._client.discovered_resource_create(
-        tid=environment, discovered_resource_id="test", values={"value1": "test1", "value2": "test2"}
+        tid=environment,
+        discovered_resource_id="invalid_rid",
+        values={"value1": "test1", "value2": "test2"},
+        discovery_resource_id="test::DiscoveryResource[agent1,key=key]",
+    )
+    assert result.code == 400
+    assert "Failed to validate argument" in result.result["message"]
+
+    result = await agent._client.discovered_resource_create(
+        tid=environment,
+        discovered_resource_id="invalid_rid",
+        values={"value1": "test1", "value2": "test2"},
     )
     assert result.code == 400
     assert "Failed to validate argument" in result.result["message"]
 
+    result = await agent._client.discovered_resource_create(
+        tid=environment,
+        discovered_resource_id="test::Resource[agent1,key=key]",
+        values={"value1": "test1", "value2": "test2"},
+        discovery_resource_id="invalid_rid",
+    )
+    assert result.code == 400
+    assert "Failed to validate argument" in result.result["message"]
```

Comment:
Shouldn't we assert on which argument it's failing? The error message might be about some other argument than the one we are expecting.